programs/summerofcode/2024.md

```diff
@@ -31,6 +31,25 @@ You can find the project ideas from previous year [here](./2023.md).
 
 ## Ideas
 
+#### Kubernetes Gateway API
+
+#### Design and develop notification mechanism for ingress2gateway.
+
+- Description: [Gateway API](https://gateway-api.sigs.k8s.io/) is the new generation of Kubernetes Networking APIs. Ingress2gateway is a tool to help users migrate and convert Ingress, and implementation-specific configurations to Gateway API. As we add more conversion logic, and onboard more implementations (essentially Ingress and Gateway controllers like Istio, Kong, etc.), logs become insufficient to report conversion results, warnings and crucial messages regarding the conversion. To improve the tool usability, we need to enable users to know exactly the original resource/s -> new resource/s mapping, PLUS what the tool couldn't convert so they could make the final touches before applying the new resources to the cluster.
+
+  A few of the notification types we'll need to support:
+  - Overview of original resources and the equivalent resources to which they have been converted
+  - Warnings about partial conversions (partial fields within a resource)
+  - Warnings about fields/logics not supported in Gateway API
+
+- Expected Outcome: A new notification mechanism to notify users conversion results, warnings and crucial messages regarding their conversion has been added and is being used across implementations supported by the tool.
+- Recommended Skills: Go, Kubernetes
+- Expected project size: medium
```

Comment:
@LiorLieberman

Although I am not an expert with the Gateway API, the project seemed larger to me. Just wanted to give you a heads up. We can still add this idea as is, if you prefer that.

Reply:
Thanks @aliok! 

I think it will sit between medium to large (considering tests as well) 

350 hours sounds a bit big to me but its the first time I am submitting a project proposal so I am not sure whats the impact of changing this? Candidates will see 350 instead of 175? 

@mlavacca thoughts?

Reply:
I think that medium fits well. I see the investigation and the design phases as pretty limited in size. The biggest part is the implementation and the addition of a comprehensive test suite.